tools/system_libs.py

```diff
@@ -1105,6 +1105,16 @@ class libasan_rt_wasm(SanitizerLibrary):
   src_dir = ['system', 'lib', 'compiler-rt', 'lib', 'asan']
 
 
+# If main() is not in EXPORTED_FUNCTIONS, it may be dce'd out. This can be
+# confusing, so issue a warning.
+def warn_on_unexported_main(symbolses):
+  if '_main' not in shared.Settings.EXPORTED_FUNCTIONS:
+    for symbols in symbolses:
+      if 'main' in symbolses[0].defs:
```

Comment:
`symbolses[0].defs`? Not `symbols.defs`?

(also I can't help but hear `symbolses` in a Gollum voice...)

Reply:
Ah thanks, that was a copy-paste error...

Yes, it's an intentional lotr reference ;)